### PR TITLE
[go_router] Fixes issue so that the parseRouteInformationWithContext …

### PR DESCRIPTION
**Pull request by chunhtai**
…can handle non-http Uris.

fixes https://github.com/flutter/flutter/issues/148389

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] page, which explains my responsibilities.
- [ ] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [ ] I signed the [CLA].
- [ ] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I [linked to at least one issue that this PR fixes] in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: packages/go_router/lib/src/path_utils.dart [hunk 1]
```diff
@@ -119,41 +119,6 @@ String concatenatePaths(String parentPath, String childPath) {
   return '${parentPath == '/' ? '' : parentPath}/$childPath';
 }
 
-/// Normalizes the location string.
-String canonicalUri(String loc) {
```
Comment (chunhtai) [inline packages/go_router/lib/src/path_utils.dart:123]: The logic here is moved to inline in parseRouteInformationWithContext

Comment (hangyujin) [reply packages/go_router/lib/src/path_utils.dart:123]: where is `parseRouteInformationWithContext` ? is it inside go_router?

Comment (chunhtai) [reply packages/go_router/lib/src/path_utils.dart:123]: yes it is a method in GoRouteInformationParser

Comment (hangyujin) [reply packages/go_router/lib/src/path_utils.dart:123]: oh you mean parseRouteInformationWithDependencies

**Review by hangyujin — APPROVED** [review]
LGTM

**Pull request merged**